src/components/clients/ClientForm.js

```diff
@@ -68,42 +57,47 @@ class ClientForm extends React.Component {
 
   render() {
     let authRedirectUrls = [];
-    for (let i = 0; i < this.state.authRedirectUrlsCount; i++) {
+    for (let i = 0; i < this.state.authRedirectUrlsInputCount; i++) {
       authRedirectUrls.push(
-        <input value={this.state.authRedirectUrls[i]} onFocus={this.onFocus}
-               onChange={this.onRedirectUrlChange} id={i} key={i} type="text"
+        <input value={this.state.authRedirectUrls[i]}
+               onChange={this.onRedirectUrlChange} id={i} key={i}
                name="authRedirectUrls" className="form-control"/>
       );
     }
 
     let authFailureRedirectUrls = [];
-    for (let i = 0; i < this.state.authFailureRedirectUrlsCount; i++) {
+    for (let i = 0; i < this.state.authFailureRedirectUrlsInputCount; i++) {
       authFailureRedirectUrls.push(
         <input value={this.state.authFailureRedirectUrls[i]}
-               onFocus={this.onFocus}
-               onChange={this.onRedirectUrlChange} id={i} key={i} type="text"
+               onChange={this.onRedirectUrlChange} id={i} key={i}
                name="authFailureRedirectUrls" className="form-control"/>
       );
     }
 
     return (
       <form className="container" onSubmit={this.onSubmit}>
         <div className="form-group">
-          <label className="control-label">Client name</label>
-          <input value={this.state.clientName} onChange={this.onChange}
-                 type="text" name="clientName" className="form-control"/>
-          <label className="control-label">Auth redirect URLs</label>
-          <button onClick={this.onClick} name="authRedirectUrlsCount"
-                  className="btn">
-            <span className="glyphicon glyphicon-plus"></span>
-          </button>
-          {authRedirectUrls}
-          <label className="control-label">Auth failure redirect URLs</label>
-          <button onClick={this.onClick} name="authFailureRedirectUrlsCount"
-                  className="btn">
-            <span className="glyphicon glyphicon-plus"></span>
-          </button>
-          {authFailureRedirectUrls}
+          <fieldset onChange={this.onChange}>
+            <label className="control-label">Client name</label>
+            <input value={this.state.clientName}
+                   name="clientName" className="form-control"/>
+          </fieldset>
+          <fieldset>
```

Comment:
nit: I would have used `<legend>` in the `<fieldset>`s instead of a label; a label usually needs to be paired with an input (or textarea).

That would also mean 1 fieldset per list of URLs.